### PR TITLE
feat(controls): add resin tagging to toolbar elements

### PR DESCRIPTION
**Pull request by kduncanhsu**
## Summary                                                                                                                                                                                                                     
  - Add `data-resin-target` attributes to 31 interactive toolbar elements across document, media, settings, and 3D controls                                                                                                    
  - Enables click tracking via the resin analytics library which reads DOM `data-resin-*` attributes                                                                                                                             
  - No code logic changes, no new dependencies, no test changes required                                                                                                                                                       
                            
resin console logger:
<img width="387" height="92" alt="Screenshot 2026-05-08 at 2 37 27 PM" src="https://github.com/user-attachments/assets/a3f564ab-3601-49c1-a013-f033861608e7" />
                                                                                                                                                                                                     
  ## Details                                                                                                                                                                                                                   
  Previously only 4 annotation buttons had resin tagging. This PR instruments all remaining interactive toolbar elements:

  | Category | Targets Added |
  |----------|--------------|
  | Document/Image | `fullscreen`, `findBar`, `thumbnails`, `rotate` |
  | Zoom | `zoomIn`, `zoomOut` |
  | Page Navigation | `pagePrevious`, `pageNext`, `pageNumberOpen`, `pageNumberInput` |
  | Color Picker | `colorPickerOpen`, `colorSelect` |
  | Media Player | `playPause`, `skipBackward`, `skipForward`, `volumeToggle`, `volumeSlider`, `timeScrubber`, `subtitlesToggle`, `mediaFullscreen` |
  | Settings | `settingsOpen`, `settingsMenuItem`, `settingsRadioItem`, `settingsCheckbox`, `settingsDropdown`, `settingsBack` |
  | 3D Controls | `vrToggle`, `resetView`, `animationClips`, `rotateAxisLeft`, `rotateAxisRight` |

  ## Test plan
  - [x] TypeScript compilation passes
  - [x] All 1508 related tests pass
  - [x] ESLint passes with no warnings
  - [ ] Verify resin events fire correctly in a staging environment with resin enabled

### CODE REVIEW

File: src/lib/viewers/controls/media/TimeControls.tsx [hunk 1]
```diff
@@ -66,6 +66,7 @@ export default function TimeControls({
             <DurationLabels currentTime={currentTime} durationTime={durationTime} />
             <SliderControl
                 className="bp-TimeControls-slider"
+                data-resin-target="timeScrubber"
```
Comment (jmcbgaston) [inline src/lib/viewers/controls/media/TimeControls.tsx:69]: Silent-failure risk: `data-resin-target` is being passed to a React component (`SliderControl`), not a DOM element. It works today because `SliderControl` spreads `{...rest}` onto its root `<div>` ([SliderControl.tsx:135](https://github.com/box/box-content-preview/blob/master/src/lib/viewers/controls/slider/SliderControl.tsx#L135)), but that's an invisible contract — if anyone ever stops spreading rest (or reorders props), the attribute silently disappears from the DOM. No type error, no test failure, just a gap in the analytics dashboard weeks later.

Low-cost mitigations:
1. Make `data-resin-target` an explicit prop on `SliderControl` applied directly to the rendered element.
2. Add a one-line assertion in the `TimeControls` test: `expect(container.querySelector('[data-resin-target="timeScrubber"]')).toBeInTheDocument()`.

Not blocking — just the class of bug that's hardest to notice.

File: src/lib/viewers/controls/media/MediaFullscreenToggle.tsx [hunk 2]
```diff
@@ -13,5 +13,5 @@ export default function MediaFullscreenToggle(props: Props): JSX.Element {
         }
     };
 
-    return <FullscreenToggle onKeyDown={handleKeydown} {...props} />;
+    return <FullscreenToggle data-resin-target="mediaFullscreen" onKeyDown={handleKeydown} {...props} />;
```
Comment (jmcbgaston) [inline src/lib/viewers/controls/media/MediaFullscreenToggle.tsx:16]: Same silent-failure risk as the `SliderControl` case in `TimeControls`, plus an extra wrinkle: `FullscreenToggle` already hardcodes `data-resin-target="fullscreen"`, and this override only works because `{...rest}` is spread *after* that attribute on the child's `<button>`. If anyone reorders props in `FullscreenToggle` so the hardcoded attribute comes after the spread, this value gets clobbered back to `"fullscreen"` — and every media fullscreen click will be logged under the wrong target name.

Suggest promoting `data-resin-target` to an explicit prop on `FullscreenToggle` so the override is visible in the component's API, or at minimum adding a test assertion here to catch the regression.

**Review by jmcbgaston — COMMENTED** [review]
Flagging a silent-failure pattern on two sites where `data-resin-target` is passed to a React component rather than a DOM element. Both work today but depend on invisible prop-forwarding contracts — if that ever breaks, analytics silently stop firing with no test or type error to catch it. Inline comments on each site.

File: src/lib/viewers/controls/media/TimeControls.tsx (hunk 1 above)
Comment (kduncanhsu) [reply src/lib/viewers/controls/media/TimeControls.tsx:69]: Good callout on defensive coding. That said, passing data-resin-target through component props via {...rest} is actually the established pattern in this codebase — the 4 existing resin tags all do exactly this (passed to AnnotationsButton, which spreads {...rest} onto its <button>).

Both SliderControl and MediaToggle explicitly type their props as extending React.HTMLAttributes / React.ButtonHTMLAttributes, so the {...rest} spread is part of their intentional public API (accepting and forwarding arbitrary HTML attributes), not an accidental implementation detail.

That said, happy to add test assertions for the cases that pass through component boundaries if you feel strongly. Let me know.

File: src/lib/viewers/controls/media/MediaFullscreenToggle.tsx (hunk 2 above)
Comment (kduncanhsu) [reply src/lib/viewers/controls/media/MediaFullscreenToggle.tsx:16]: same as above

**Review by ahorowitz123 — APPROVED** [review]
